Source/Core/DolphinQt/HotkeyScheduler.cpp

```diff
@@ -537,57 +538,15 @@ void HotkeyScheduler::Run()
       Config::SetCurrent(Config::GFX_STEREO_CONVERGENCE,
                          std::min(stereo_convergence + 5, Config::GFX_STEREO_CONVERGENCE_MAXIMUM));
 
-    // Freelook
-    static float fl_speed = 1.0;
-
+    // Free Look
     if (IsHotkey(HK_FREELOOK_TOGGLE))
     {
-      const bool new_value = !Config::Get(Config::GFX_FREE_LOOK);
-      Config::SetCurrent(Config::GFX_FREE_LOOK, new_value);
-      OSD::AddMessage(StringFromFormat("Freelook: %s", new_value ? "Enabled" : "Disabled"));
+      const bool new_value = !Config::Get(Config::FREE_LOOK_ENABLED);
+      Config::SetCurrent(Config::FREE_LOOK_ENABLED, new_value);
+      OSD::AddMessage(StringFromFormat("Free Look: %s", new_value ? "Enabled" : "Disabled"));
     }
 
-    if (IsHotkey(HK_FREELOOK_DECREASE_SPEED, true))
-      fl_speed /= 1.1f;
-
-    if (IsHotkey(HK_FREELOOK_INCREASE_SPEED, true))
-      fl_speed *= 1.1f;
-
-    if (IsHotkey(HK_FREELOOK_RESET_SPEED, true))
-      fl_speed = 1.0;
-
-    if (IsHotkey(HK_FREELOOK_UP, true))
-      g_freelook_camera.MoveVertical(-fl_speed);
-
-    if (IsHotkey(HK_FREELOOK_DOWN, true))
-      g_freelook_camera.MoveVertical(fl_speed);
-
-    if (IsHotkey(HK_FREELOOK_LEFT, true))
-      g_freelook_camera.MoveHorizontal(fl_speed);
-
-    if (IsHotkey(HK_FREELOOK_RIGHT, true))
-      g_freelook_camera.MoveHorizontal(-fl_speed);
-
-    if (IsHotkey(HK_FREELOOK_ZOOM_IN, true))
-      g_freelook_camera.Zoom(fl_speed);
-
-    if (IsHotkey(HK_FREELOOK_ZOOM_OUT, true))
-      g_freelook_camera.Zoom(-fl_speed);
-
-    if (IsHotkey(HK_FREELOOK_INCREASE_FOV_X, true))
-      g_freelook_camera.IncreaseFovX(g_freelook_camera.GetFovStepSize());
-
-    if (IsHotkey(HK_FREELOOK_DECREASE_FOV_X, true))
-      g_freelook_camera.IncreaseFovX(-1.0f * g_freelook_camera.GetFovStepSize());
-
-    if (IsHotkey(HK_FREELOOK_INCREASE_FOV_Y, true))
-      g_freelook_camera.IncreaseFovY(g_freelook_camera.GetFovStepSize());
-
-    if (IsHotkey(HK_FREELOOK_DECREASE_FOV_Y, true))
-      g_freelook_camera.IncreaseFovY(-1.0f * g_freelook_camera.GetFovStepSize());
-
-    if (IsHotkey(HK_FREELOOK_RESET, true))
-      g_freelook_camera.Reset();
+    FreeLook::UpdateInput();
```

Comment:
Also, not sure you've seen that FreeLook won't run if Hotkeys are disabled, there is a
`if (!HotkeyManagerEmu::IsEnabled()) return` condition above, I'm guessing it should run independently of that setting.

Reply:
I did miss that hotkeys can be disabled.  But as I mention in this PR, this was a simple stop-gap to get things merged more quickly.  Long term I'd expect there to be a separate input thread that calls both hotkey and FreeLook systems.

I'd personally like to add a new "background input" setting specifically for FreeLook because while I'd assume users would want it to operate it similarly to the game, they might not.  I'd also add an "Input Sources" button like in the main controller window.  It's confusing to me if users have to go somewhere else to modify those settings for FreeLook.